playbooks/roles/ocp-ovnkube-cni/tasks/networkpolicy.yml

```diff
@@ -245,97 +245,108 @@
+          kind: Project
+          metadata:
+            name: cluster
+          spec: 
+            projectRequestTemplate: 
+              name: project-request
 
-    - debug:
-        msg: "{{ create_template.stdout }}"
+    - name: Apply new custom default project template
+      kubernetes.core.k8s:
+        state: present
+        src: "{{ ovnkube_setup_directory }}/template.yml"
 
-    - name: Get the project.config.openshift.io/cluster resource
-      shell: oc get project.config.openshift.io/cluster -o yaml > "{{ ovnkube_setup_directory }}/default_policy_config.yaml"
+    - name: Pause for 5 minutes to apply default policy
+      pause:
+        minutes: 5
 
-    - name: Edit the project.config.openshift.io/cluster resource
-      lineinfile:
-        path: "{{ ovnkube_setup_directory }}/default_policy_config.yaml"
-        regexp: "spec: {}"
-        state: absent
+    - name: Create the new project
+      shell: oc new-project test-default-network-policy
 
-    - name: Edit the project configuration resource
-      blockinfile:
-        block: "{{ lookup('file', 'default_policy_config.txt') }}"
-        path: "{{ ovnkube_setup_directory }}/default_policy_config.yaml"
-    
-    - name: Apply changes project configuration resource
-      shell: oc apply -f "{{ ovnkube_setup_directory }}/default_policy_config.yaml"
-      register: create_config
+    - name: Check if default network policy has been added to the new project
+      shell: oc get networkpolicy -n test-default-network-policy
+      register: network_policy
+      until: '"allow-from-same-namespace" in network_policy.stdout'
 
-    - name: Create a namespace
+    - name: Delete the project test-default-network-policy
       kubernetes.core.k8s:
-        state: present
-        definition:
-          apiVersion: v1
-          kind: Namespace
-          metadata:
-            labels:
-              pod-security.kubernetes.io/audit: privileged
-              pod-security.kubernetes.io/enforce: privileged
-              pod-security.kubernetes.io/warn: privileged
-              security.openshift.io/scc.podSecurityLabelSync: "false"
-            name: test-3
-
-    - name: Confirm that the network policy objects in the new project template exist in the new project
-      shell: oc get networkpolicy -n test-3
-      register: confim_op
-      failed_when: '"allow-from-same-namespace" not in confim_op.stdout'
- 
-    - debug:
-        msg: "{{ confim_op.stdout_lines }}"
+        api_version: project.openshift.io/v1
+        state: absent
+        kind: Project
+        name: test-default-network-policy
 
-    - name: Delete template
-      shell: oc delete -f "{{ ovnkube_setup_directory }}/template.yaml" -n openshift-config
-      ignore_errors: yes
+    - name: Check if the project gets deleted
+      shell: oc get project | grep test-default-network-policy | wc -l
+      register: test_project
+      until: test_project.stdout|int == 0
+      retries: 15
+      delay: 30
```

Comment:
@varad-ahirwadkar I think wait is available in k8s module for deletion on all resources https://docs.ansible.com/ansible/latest/collections/kubernetes/core/k8s_module.html#parameter-wait.
Could you please try combining these two tasks into one? We won't need oc shell commands everywhere.

Reply:
Sure. I have combined these two tasks.